utility/types/tx_fifo_mempool.go

```diff
@@ -0,0 +1,108 @@
+package types
+
+import (
+	"sync"
+
+	"github.com/pokt-network/pocket/shared/crypto"
+	"github.com/pokt-network/pocket/shared/mempool"
+)
+
+var _ mempool.TXMempool = &txFIFOMempool{}
+
+type txFIFOMempool struct {
+	g                *mempool.GenericFIFOSet[string, []byte]
+	m                sync.Mutex
+	txCount          uint32 // current number of transactions in the mempool
+	txsBytesTotal    uint64 // current sum of all transactions' sizes (in bytes)
+	maxTxsBytesTotal uint64 // maximum total size of all txs allowed in the mempool
+}
+
+// AddTx adds a tx to the mempool
+func (t *txFIFOMempool) AddTx(tx []byte) error {
+	if err := t.g.Push(tx); err != nil {
+		return ErrDuplicateTransaction()
+	}
+	return nil
+}
+
+// Clear clears the mempool
+func (t *txFIFOMempool) Clear() {
+	t.g.Clear()
```

Comment:
Same as before. My inutition just makes me think:

1. Lock outer structure
2. Clear inner structure
   2.1. Capture inner lock
   2.2. Unlock inner lock
3. Clear outer strcture
4. Unlock outer structure

Reply:
My 🚂 of 🤔 is:
1. Do what the generic implementation does, 
2. Do what's special to this implementation

Basically this:  https://github.com/pokt-network/pocket/pull/437/commits/b875b37f4a85499120adfccd8b3f99faea654972

Reply:
Won't push back against this since it doesn't matter.

---

Going to share my personal 🚄  of 🤔 thought:

1. In polymorphism (when a subclass inherits from a superclass), we usually do:

```
super.constructor()
self.constructioner()
```

2. In this case, where the generic implementation is an internal field, I'm thinking:

```
self.clearFields()
attributes.clearFields()
```

Reply:
Well, it's not a rule, you can call the super before or after depending on how the subclass needs to behave.

If for example `super.constructor()` prints something that is dependent on some internal state (accessible and modifiable by the `self`, it makes sense for the `self.constructioner()` to happen before.